src/gtc/cuir/kernel_fusion.py

```diff
@@ -0,0 +1,63 @@
+# -*- coding: utf-8 -*-
+#
+# GTC Toolchain - GT4Py Project - GridTools Framework
+#
+# Copyright (c) 2014-2021, ETH Zurich
+# All rights reserved.
+#
+# This file is part of the GT4Py project and the GridTools framework.
+# GT4Py is free software: you can redistribute it and/or modify it under
+# the terms of the GNU General Public License as published by the
+# Free Software Foundation, either version 3 of the License, or any later
+# version. See the LICENSE.txt file at the top-level directory of this
+# distribution for a copy of the license or check <https://www.gnu.org/licenses/>.
+#
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+from typing import Set
+
+from eve import NodeTranslator
+
+from . import cuir
+
+
+class FuseKernels(NodeTranslator):
```

Comment:
Please explain what the optimization does and when it can be applied

Reply:
Also added some tests and adapted the transformation to make sure it does not over-fuse as GT did before on parallel computations with vertical offsets.